### PR TITLE
E2E test: hook up with verification

### PR DESCRIPTION
**Pull request by chaodaiG**
Also refactor integration test to be able to share more code with e2e test:
- Separate server setup from data seeding
- Authorized app provider backed by db instead of memory

### CODE REVIEW

**Comment by chaodaiG** [comment]
@sethvargo , PTAL

File: internal/integration/helpers.go [hunk 1]
```diff
@@ -259,21 +234,57 @@ func NewTestServer(tb testing.TB, exportPeriod time.Duration) (*serverenv.Server
...
+		if err := authorizedappdb.New(db).InsertAuthorizedApp(context.Background(), &authorizedappmodel.AuthorizedApp{
+			AppPackageName: "com.example.app",
+			AllowedRegions: map[string]struct{}{
+				"TEST": {},
+			},
+			AllowedHealthAuthorityIDs: map[int64]struct{}{
+				1: {},
+			},
+
+			// TODO: hook up verification and revision
```
Comment (sethvargo) [inline internal/integration/helpers.go:282]: False is the default value, which means this _is_ hooked up.

**Review by sethvargo — APPROVED** [review]
/lgtm

One comment seems outdated.

**Pull request merged**